lib/services/sshAdapter.js

```diff
@@ -0,0 +1,64 @@
+"use strict";
+
+var Bluebird = require("bluebird");
+
+function SshAdapter (ec2) {
+	var SSH_POLLING_INTERVAL = 30; /*Time in Seconds*/
+
+	this.checkIfSshable = Bluebird.method(function (instanceID) {
+		return this.canSsh(instanceID)
+		.then(function (sshable) {
+			if (sshable) {
+				return Bluebird.resolve();
+			}
+			else {
+				return Bluebird.delay(SSH_POLLING_INTERVAL * 1000)
+				.then(this.checkIfSshable.bind(this, instanceID));
+			}
+		}.bind(this));
```

Comment:
While this is a valid thing to do, I find that it tends to make the code more difficult to read since it changes the normal receiver of the function and does so in a way that is not localized to the function body. I prefer using something like `var self = this;` in the outer scope and then using `self` rather than `this` throughout the function body.